### DISCUSSION TITLE
Kermit for Newrelic

### DISCUSSION THREAD

**Discussion by ndesai-newrelic** [touchlab/Kermit#236] [Ideas]
is there any plan to add Kermit for Newrelic?

**Comment by kpgalligan** [comment]
Generally speaking, we don't plan to support anything we're (Touchlab) not actively using. We do take PR's for some things.

However, we were working with New Relic on a project about a year ago. For crash reporting specifically, New Relic's client does not support the custom exception with function pointers capability we'd need for symbolicated stack reports, or at least didn't back then.

Sentry was the same situation back in 2019, but had since changed their client, and somebody submitted a PR.

If New Relic supports creating custom exception reports with function pointers (not just strings), it may be possible for that.

As for just "logging", It would be relatively trivial to implement a LogWriter that logged out to any client. That's pretty simple. The crash reporting is generally much more complex to set up.

**Comment by ndesai-newrelic** [comment]
@kpgalligan Now we have method where you can send custom exceptions

**Reply by kpgalligan** [reply]
@ndesai-newrelic Looking at your username, are you with newrelic? Do you have any more details? At this point, it would go into [CrashKiOS](https://github.com/touchlab/CrashKiOS), but that's an implementation detail.

**Reply by ndesai-newrelic** [reply]
@kpgalligan  I am with newrelic, we are planning to support KMM in next Quarter, I can provide you  method which you can use to send data to newrelic.
